gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -471,6 +493,77 @@ private static void setupConjurePythonProject(
         }
     }
 
+    private static void setupGenericConjureProjects(
+            Project project,
+            Function<String, GeneratorOptions> getGenericOptions,
+            Task compileConjure,
+            Task compileIrTask,
+            Configuration conjureGeneratorsConfiguration) {
+        // Validating that each subproject has a corresponding generator.
+        // We do this in afterEvaluate to ensure the configuration is populated.
+        project.afterEvaluate(p -> {
+            Map<String, Dependency> generators = conjureGeneratorsConfiguration
+                    .getAllDependencies()
+                    .stream()
+                    .collect(Collectors.toMap(dependency -> {
+                        Preconditions.checkState(dependency.getName().startsWith(CONJURE_GENERATOR_DEP_PREFIX),
+                                "Generators should start with '%s' according to conjure RFC 002, "
+                                        + "but found name: '%s' (%s)",
+                                CONJURE_GENERATOR_DEP_PREFIX, dependency.getName(), dependency);
+                        return dependency.getName().substring(CONJURE_GENERATOR_DEP_PREFIX.length());
+                    }, Function.identity()));
+
+            project.getChildProjects().entrySet().stream()
+                    .forEach(entry -> {
+                        String subprojectName = entry.getKey();
+                        Project subproject = entry.getValue();
+                        String conjureLanguage = extractSubprojectLanguage(project.getName(), subprojectName);
+                        if (!FIRST_CLASS_GENERATOR_PROJECT_NAMES.contains(conjureLanguage)
+                                && !generators.containsKey(conjureLanguage)) {
+                            throw new RuntimeException(String.format("Discovered subproject %s without corresponding "
+                                            + "generator dependency with name '%s'",
+                                    subproject.getPath(), ConjurePlugin.CONJURE_GENERATOR_DEP_PREFIX + subprojectName));
+                        }
+                    });
+        });
+
+        project.getChildProjects().entrySet().stream()
+                .filter(e -> FIRST_CLASS_GENERATOR_PROJECT_NAMES.stream()
```

Comment:
This whole afterEvaluate block is just working on these generic subprojects, not the first-class ones... Maybe just hoist a variable like genericSubprojects to do this once? e.g.

```
Maps.filterKeys(project.getChildProjects(), key -> !FIRST_CLASS_GENERATOR_PROJECT_NAMES.contains(key));
```